### PR TITLE
Remove Ruff and Uvicorn in Wasm env

### PR DESCRIPTION
**Pull request by whitphx**
## Description

Related to #7134

* Exclude some requirements for Lite. It should help reduce the init loading time and the loaded payload size.
    * I originally intended to exclude all removable packages, but found some code modifications could be needed per package, so I changed my mind to do it one by one.
    * In this PR, `ruff` and `uvicorn` is the target.
* `ruff` could be simply removed: 9175e9c
* Removing `uvicorn` needed some code changes in the following commits. I moved the `Server` class and the `start_server` function which depend on `uvicorn` to a separate module `http_server.py`, and made `blocks.py` import it only in non-Wasm mode.

### CODE REVIEW

**Comment by whitphx** [comment]
This CI failure should be fixed by #7770

File: test/test_blocks.py [hunk 1]
```diff
@@ -351,18 +352,19 @@ async def test_run_without_launching(self):
```
Comment (whitphx) [inline test/test_blocks.py:352]: This test case was introduced in https://github.com/gradio-app/gradio/pull/3089, and became unnecessary in https://github.com/gradio-app/gradio/pull/5267 ?

Comment (freddyaboulton) [reply test/test_blocks.py:352]: Yea I think we can delete this

**Review by freddyaboulton — APPROVED** [review]
Changes look good to me @whitphx !

File: test/test_blocks.py (hunk 1 above)
Comment (whitphx) [reply test/test_blocks.py:352]: Thanks!

Comment (whitphx) [reply test/test_blocks.py]: Thanks, deleted 👍

**Comment by whitphx** [comment]
Thanks!
This was conflicted with https://github.com/gradio-app/gradio/pull/7684 and resolved.
Merge this PR once the CI passes.

**Comment by whitphx** [comment]
@abidlabs Did it, thanks!

**Pull request merged**